oauth.go

```diff
@@ -194,8 +196,8 @@ func newConsumer(consumerKey string,
 //        see the documentation for ServiceProvider for how to create this.
 //
 func NewConsumer(consumerKey string, consumerSecret string,
```

Comment:
It appears that a number of people are using this constructor as-is (https://github.com/search?q=oauth.NewConsumer&type=Code&utf8=%E2%9C%93), and I'd prefer not to break them if possible.

A couple things we could do here:
1. Add a second constructor, which takes an httpClient.  As you can see someone has already done this to support different "signers" in NewRSAConsumer.  Perhaps we could even go as far as to expose a NewConsumerExt(), which takes both an "httpClient" and a "signer".
2. We could also change httpClient to be a public member of consumer (by changing the name to HttpClient), and then anyone would be free to override it themselves (the test already does this, but it only because it cheats by being in the same package).

Perhaps there are other good options too.

I don't have a terribly strong preference, but #1 seems like it might be the best bed ... it does add yet another constructor, but at least it's clearly defined how it will work.  I'm a little worried that if we expose HttpClient directly someone might shoot themselves in the foot one day (e.g. by overriding it while a request is in flight).